### ISSUE TITLE
XDMFFile.read_mvc_size_t broken since: f3b20b410a7925030434807e9e22f8682b774485

### ISSUE THREAD

**Issue by jorgensd** [FEniCS/dolfinx#818]
Minimal example working before f3b20b410a7925030434807e9e22f8682b774485,
```
from dolfinx.io import XDMFFile
from dolfinx import MPI
import dolfinx 

mesh_file = XDMFFile(MPI.comm_world, "cf.xdmf")
mesh = mesh_file.read_mesh(dolfinx.cpp.mesh.GhostMode.none)
mvc = mesh_file.read_mvc_size_t(mesh, "name_to_read")
```
with the following mesh:
```
<Xdmf Version="3.0"><Domain><Grid Name="Grid"><Information Name="Information" Value="0">
<![CDATA[<main />]]>
</Information><Geometry GeometryType="XYZ"><DataItem DataType="Float" Dimensions="7 3" Format="XML" Precision="8">
2.500000000000000e-01 -2.165063509461097e-01 0.000000000000000e+00
0.000000000000000e+00 2.165063509461097e-01 0.000000000000000e+00
-2.500000000000000e-01 -2.165063509461097e-01 0.000000000000000e+00
-5.000000000000000e-01 -5.000000000000000e-01 0.000000000000000e+00
5.000000000000000e-01 -5.000000000000000e-01 0.000000000000000e+00
5.000000000000000e-01 5.000000000000000e-01 0.000000000000000e+00
-5.000000000000000e-01 5.000000000000000e-01 0.000000000000000e+00
</DataItem></Geometry><Topology NumberOfElements="8" TopologyType="Triangle"><DataItem DataType="Int" Dimensions="8 3" Format="XML" Precision="8">
0 1 2
3 0 2
1 5 6
0 5 1
4 0 3
4 5 0
2 1 6
3 2 6
</DataItem></Topology><Attribute AttributeType="Scalar" Center="Cell" Name="name_to_read"><DataItem DataType="Int" Dimensions="8" Format="XML" Precision="8">
1
2
2
2
2
2
2
2
</DataItem></Attribute></Grid></Domain></Xdmf>
```
In versions later than this commit, it returns:
```
Traceback (most recent call last):
  File "mesh_MWE.py", line 7, in <module>
    mvc = mesh_file.read_mvc_size_t(mesh, "name_to_read")
  File "/home/shared/src/dolfinx/python/dolfinx/io.py", line 219, in read_mvc_size_t
    return self._cpp_object.read_mvc_size_t(mesh, name)
RuntimeError: Cannotfind entity in map. Error reading mesh::MeshValueCollection

```
when ran in parallel

**Comment by jorgensd** [comment]
Minimal code for reproducing the mesh with a parallel supported format:
```
import pygmsh
import meshio
geom = pygmsh.built_in.Geometry()
# NOTE: Make a line mesh to illustrate meshio issue with paraview
rect = geom.add_rectangle(0, 1, 0, 1, 0, lcar=0.5)
geom.add_physical(rect.surface, 1)

msh = pygmsh.generate_mesh(geom)

for key in msh.cell_data_dict["gmsh:physical"].keys():
    if key == "triangle":
        triangle_data = msh.cell_data_dict["gmsh:physical"][key]
for cell in msh.cells:
    if cell.type == "triangle":
        triangle_cells = cell.data

triangle_mesh =meshio.Mesh(points=msh.points,
                           cells=[("triangle", triangle_cells)],
                           cell_data={"name_to_read":[triangle_data]})
meshio.xdmf.write("cf.xdmf", triangle_mesh)
``` 
Just adding it here for having a complete story, as Garth is currently rewriting the MVC IO

**Comment by francesco-ballarin** [comment]
A variation of @jorgensd 's example to show a (similar?) issue when marking facets
```
import pygmsh
from dolfinx import Mesh, MeshValueCollection, MPI
from dolfinx.cpp.mesh import CellType, GhostMode

geom = pygmsh.built_in.Geometry()
rect = geom.add_rectangle(0, 1, 0, 1, 0, lcar=0.5)
geom.add_physical(rect.surface, 1)
geom.add_physical(rect.lines[0], 2)

pygmsh_mesh = pygmsh.generate_mesh(geom)

mesh = Mesh(MPI.comm_world, CellType.triangle, pygmsh_mesh.points[:, :2],
            pygmsh_mesh.cells_dict["triangle"], [], GhostMode.none)

subdomains_mvc = MeshValueCollection("size_t", mesh, mesh.topology.dim, pygmsh_mesh.cells_dict["triangle"],
                                     pygmsh_mesh.cell_data_dict["gmsh:physical"]["triangle"])

boundaries_mvc = MeshValueCollection("size_t", mesh, mesh.topology.dim - 1, pygmsh_mesh.cells_dict["line"],
                                     pygmsh_mesh.cell_data_dict["gmsh:physical"]["line"])
# The previous line fails with
#  File "dolfinx/mesh.py", line 70, in MeshValueCollection
#    return mvc(mesh, dim, topology_data, value_data)
# RuntimeError: Entity not found in the mesh.
```

Is this expected (due to the current rewrite)? If not, do you want me to move this to a new issue instead?

**Comment by garth-wells** [comment]
Classes since removed. Functionality will return with MeshTags.
